assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,561 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:EditBuildConfigController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('EditBuildConfigController', function ($scope, $routeParams, DataService, ProjectsService, $filter, ApplicationGenerator, Navigate, $location, AlertMessageService, SOURCE_URL_PATTERN) {
+
+    $scope.projectName = $routeParams.project;
+    $scope.buildConfig = null;
+    $scope.alerts = {};
+    $scope.emptyMessage = "Loading...";
+    $scope.sourceURLPattern = SOURCE_URL_PATTERN;
+    $scope.options = {};
+    $scope.buildFromTypes = [
+      {
+        "id": "ImageStreamTag",
+        "title": "Image Stream Tag"
+      },
+      {
+        "id": "DockerImage",
+        "title": "Docker Image Link"
+      }
```

Comment:
we also allow ImageStreamImage in the from: https://github.com/openshift/origin/blob/master/pkg/build/api/validation/validation.go#L327-L365